src/main/java/org/hyperledger/fabric/sdk/BlockDeserializer.java

```diff
@@ -80,10 +79,14 @@ EnvelopeDeserializer getData(int index) throws InvalidProtocolBufferException {
 
     }
 
-    byte[] getTransActionsMetaData() {
-
-        return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+    byte[] getTransActionsMetaData() throws InvalidProtocolBufferRuntimeException {
 
+        List<ByteString> metadataList = block.getMetadata().getMetadataList();
+        if (metadataList != null && metadataList.size() > Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE) {
+            return block.getMetadata().getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
+        } else {
+            throw new InvalidProtocolBufferRuntimeException(new InvalidProtocolBufferException("empty meta data from fabric"));
+        }
```

Comment:
Since you've already got `metadataList`, you could pull the value from that with `metadataList.get(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE)`.

Or alternatively, something like:

```suggestion
        Common.BlockMetadata metadata = block.getMetadata();
        if (Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE >= metadata.getMetadataCount()) {
            throw new InvalidProtocolBufferRuntimeException(new InvalidProtocolBufferException("Missing TRANSACTION_FILTER in block metadata"));
        }

        return metadata.getMetadata(Common.BlockMetadataIndex.TRANSACTIONS_FILTER_VALUE).toByteArray();
```